apps/login/cypress/integration/login.cy.ts

```diff
@@ -0,0 +1,115 @@
+import { addStub, removeStub } from "../support/mock";
+
+describe("login", () => {
+  beforeEach(() => {
+    removeStub("zitadel.session.v2alpha.SessionService", "CreateSession");
+    addStub("zitadel.session.v2alpha.SessionService", "CreateSession", {
+      data: {
+        details: {
+          sequence: 859,
+          changeDate: "2023-07-04T07:58:20.126Z",
+          resourceOwner: "220516472055706145",
+        },
+        sessionId: "221394658884845598",
+        sessionToken:
+          "SDMc7DlYXPgwRJ-Tb5NlLqynysHjEae3csWsKzoZWLplRji0AYY3HgAkrUEBqtLCvOayLJPMd0ax4Q",
+        challenges: undefined,
+      },
+    });
+
+    removeStub("zitadel.session.v2alpha.SessionService", "GetSession");
+    addStub("zitadel.session.v2alpha.SessionService", "GetSession", {
+      data: {
+        session: {
+          id: "221394658884845598",
+          creationDate: "2023-07-04T07:58:20.026Z",
+          changeDate: "2023-07-04T07:58:20.126Z",
+          sequence: 859,
+          factors: {
+            user: {
+              id: "123",
+              loginName: "john@zitadel.com",
+            },
+            password: undefined,
+            passkey: undefined,
+            intent: undefined,
+          },
+          metadata: {},
+          domain: "localhost",
+        },
+      },
+    });
+  });
+  it("should redirect a user with password authentication to /password", () => {
+    removeStub(
+      "zitadel.user.v2alpha.UserService",
+      "ListAuthenticationMethodTypes"
+    );
+    addStub(
+      "zitadel.user.v2alpha.UserService",
+      "ListAuthenticationMethodTypes",
+      {
+        data: {
+          authMethodTypes: [1], // 1 for password authentication
+        },
+      }
+    );
```

Comment:
Let's move stubs to beforeEach hooks. Also in verify.cy.ts.

We can nest contexts with describe blocks.
maybe a describe block with password and one with passwordless?
Their tests have common stubs, which we could even reuse in beforeEach.

Reply:
I've restructured the `login.cy.ts` file 👍

Reply:
I reduced the code a bit more https://github.com/zitadel/typescript/pull/33/commits/1bd5ea4e2d7e702d3e0c7b0702d1b377700c0c71